File: daily_python/3536_Maximum_Product_of_Two_Digits.py
import os
import sys
from typing import List

class Solution:
    def maxProduct(self, n: int) -> int:
        v1, v2 = -1, -1 
        while n:
            curr = n % 10
            if curr > v1:
                v2 = v1
                v1 = curr
            else:
                v2 = max(v2, curr)
            n //= 10
        return v1 * v2

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            n = int(lines[0].split('=')[1].strip()[:-1])
            dataset.append(n)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for n in dataset:
        results.append(solution.maxProduct(n))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')

File: daily/2026/3536_Maximum_Product_of_Two_Digits.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int maxProduct(int n) {
        int v1 = -1;
        int v2 = -1;
        while(n){
            int curr = n % 10;
            v2 = max(min(v1, curr), v2);
            v1 = max(v1, curr);
            n /= 10;
        }
        return v1 * v2;
    }
};

int main(){
    int n = 31;
    // int n = 22;
    // int n = 124;
    Solution S;
    cout << S.maxProduct(n) << endl;
    return 0;
}